src/backoff.ts

```diff
@@ -19,28 +19,50 @@ export async function backOff<T>(
 
 class BackOff<T> {
   private attemptNumber = 0;
+  private isAborted = false;
 
   constructor(
     private request: () => Promise<T>,
     private options: IBackOffOptions
   ) {}
 
+  private _cancelWhenAborted() {
+    const { signal } = this.options;
+
+    if (!signal) return;
+
+    if (signal.aborted) {
+      this.isAborted = true;
+      return;
+    }
+
+    signal.addEventListener(
+      "abort",
+      () => {
+        this.isAborted = true;
+      },
+      { once: true }
+    );
+  }
+
   public async execute(): Promise<T> {
-    while (!this.attemptLimitReached) {
+    this._cancelWhenAborted();
+
+    while (!this.isAborted && !this.attemptLimitReached) {
       try {
         await this.applyDelay();
         return await this.request();
       } catch (e) {
         this.attemptNumber++;
         const shouldRetry = await this.options.retry(e, this.attemptNumber);
 
-        if (!shouldRetry || this.attemptLimitReached) {
+        if (!shouldRetry || this.isAborted || this.attemptLimitReached) {
           throw e;
         }
       }
     }
 
-    throw new Error("Something went wrong.");
+    throw new Error(this.isAborted ? "Retry aborted" : "Something went wrong.");
```

Comment:
my only gripe is not using [`signal.reason` if available](https://nodejs.org/api/globals.html#abortsignalreason). That would be a bit of a refactor here, but I don't think we'd want to NOT use one of the built in values of the AbortController. 

I think its reasonable that the reason would populate into the error here to be handled, as well as a separate handling of that inside the catch